Cannot locally test azul-docker-pycharm changes

Starting with bd7c321f353c05c838f7ceec400029057a0ea8f5, the instructions in the [azul-docker-pycharm](https://github.com/DataBiosphere/azul-docker-pycharm/) repo to test changes locally no longer work.

> \# To test the image in Azul, run:
> (cd ../azul && azul_docker_pycharm_version=2023.2.3-5 azul_docker_registry=localhost:5000/ make format)

```
(.venv) daniel@Crispin ~/repo/azul1 $ azul_docker_pycharm_version=2023.3.3-14 azul_docker_registry=localhost:5000/ make format
docker run \
	    --rm \
	    --volume $(python scripts/resolve_container_path.py /Users/daniel/repo/azul1):/home/developer/azul \
	    --workdir /home/developer/azul \
	    $(AZUL_DEBUG=0 python -m azul 'docker.resolve_docker_image_for_launch("pycharm")') \
	    /opt/pycharm/bin/format.sh -r -settings .pycharm.style.xml -mask '*.py' src scripts test lambdas/layer/app.py lambdas/indexer/app.py lambdas/service/app.py .flake8/azul_flake8.py environment.py deployments/anvilbox/environment.py deployments/anvildev.browser/environment.py deployments/anvildev.gitlab/environment.py deployments/anvildev.shared/environment.py deployments/anvildev/environment.py deployments/anvilprod.browser/environment.py deployments/anvilprod.gitlab/environment.py deployments/anvilprod.shared/environment.py deployments/anvilprod/environment.py deployments/daniel.local/environment.py deployments/daniel2.local/environment.py deployments/dev.gitlab/environment.py deployments/dev.shared/environment.py deployments/dev/environment.py deployments/hammerbox/environment.py deployments/prod.gitlab/environment.py deployments/prod.shared/environment.py deployments/prod/environment.py deployments/sandbox/environment.py .github/workflows/schedule.py terraform/backend.tf.json.template.py terraform/common.tf.json.template.py terraform/step_function.tf.json.template.py terraform/lambda_layer.tf.json.template.py terraform/dynamo.tf.json.template.py terraform/cloudwatch.tf.json.template.py terraform/elasticsearch.tf.json.template.py terraform/api_gateway.tf.json.template.py terraform/providers.tf.json.template.py terraform/authentication.tf.json.template.py terraform/sqs.tf.json.template.py terraform/s3.tf.json.template.py terraform/gitlab/gitlab.tf.json.template.py terraform/shared/shared.tf.json.template.py terraform/browser/browser.tf.json.template.py lambdas/indexer/.chalice/config.json.template.py lambdas/service/.chalice/config.json.template.py .github/pull_request_template.md.template.py
envhook.py: Warning: azul_docker_registry is 'localhost:5000/' but should be '122796619775.dkr.ecr.us-east-1.amazonaws.com/', you must run `source environment`
envhook.py: Warning: AZUL_DEBUG is '0' but should be '1', you must run `source environment`
envhook.py: Warning: azul_docker_registry is 'localhost:5000/' but should be '122796619775.dkr.ecr.us-east-1.amazonaws.com/', you must run `source environment`
2024-02-23 15:04:31,009    INFO MainThread azul.docker: Resolving image 'pycharm' TagImageRef(registry='docker.io', username='ucscgi', repository=('azul-pycharm',), tag='2023.3.2-11') …
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/Users/daniel/repo/azul1/src/azul/__main__.py", line 21, in <module>
    result = str(eval(expression, dict(__builtins__={}), locals))
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<string>", line 1, in <module>
  File "/Users/daniel/repo/azul1/src/azul/docker.py", line 511, in resolve_docker_image_for_launch
    ref_to_pull, manifest = resolve_docker_image_for_pull(alias)
                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/daniel/repo/azul1/src/azul/docker.py", line 542, in resolve_docker_image_for_pull
    ref_to_pull = TagImageRef.parse(config.docker_registry + str(ref))
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/Users/daniel/repo/azul1/src/azul/docker.py", line 218, in parse
    name, tag = image_ref.split(':')
    ^^^^^^^^^
ValueError: too many values to unpack (expected 2)
docker: invalid reference format.
See 'docker run --help'.
make: *** [format] Error 125
```

--- a/.github/workflows/docker-publish.yml
+++ b/.github/workflows/docker-publish.yml
@@ -3,8 +3,8 @@ name: Docker
 on:
   pull_request:
   push:
-    branches-ignore:
-      - "*-PR"
+    branches:
+      - main
 
 env:
   # The fully qualified image name of the image to build. The first component is

--- a/.github/workflows/docker-publish.yml
+++ b/.github/workflows/docker-publish.yml
@@ -78,7 +78,9 @@ jobs:
             env.azul_docker_registry,
             env.azul_docker_elasticsearch_image
           ) }}"
-          cache-to: "${{ format(
+          # The value after '&&' in a ternary operator must be truthy, and since
+          # an empty string is falsy, we use ' ' instead.
+          cache-to: "${{ github.event_name == 'pull_request' && ' ' || format(
             'type=registry,ref={0}{1}_buildcache,mode=max',
             env.azul_docker_registry,
             env.azul_docker_elasticsearch_image

--- a/README.md
+++ b/README.md
@@ -17,11 +17,24 @@ docker pull localhost:5000/docker.io/ucscgi/azul-elasticsearch:7.17.15-6
 # add a `RUN` instruction to the Dockerfile to remove the package.
 # To test the image in Azul, you will need to temporarily modify Azul's
 # `environment.py` to set the appropriate `azul_docker_images` value using the
-# full image name noted above (starting with "localhost"). With this change in
-# place, run:
-_refresh
+# full image name noted above (starting with "localhost").
+cd ../azul
+git diff
+>   diff --git a/environment.py b/environment.py
+>   index f888a1b8e..5970816ac 100644
+>   --- a/environment.py
+>   +++ b/environment.py
+>   @@ -276,7 +276,7 @@ def env() -> Mapping[str, Optional[str]]:
+>                    'ref': 'docker.io/ucscgi/azul-pycharm:2023.3.4-15'
+>                },
+>                'elasticsearch': {
+>   -                'ref': 'docker.io/ucscgi/azul-elasticsearch:7.17.18-13'
+>   +                'ref': 'localhost:5000/docker.io/ucscgi/azul-elasticsearch:7.17.18-13'
+>                },
+>                'bigquery_emulator': {
+>                    'ref': 'ghcr.io/hannes-ucsc/bigquery-emulator:azul'
 make image_manifests.json
 azul_docker_registry="" make test
-# Finally, back in your working tree for this repository:
+cd -
 make stop_registry
 ```

--- a/README.md
+++ b/README.md
@@ -15,7 +15,13 @@ docker pull localhost:5000/docker.io/ucscgi/azul-elasticsearch:7.17.15-6
 # To examine the image for vulnerabilities, browse the image in Docker Desktop.
 # If an unnecessary package is found to have critical or high vulnerabilities,
 # add a `RUN` instruction to the Dockerfile to remove the package.
-# To test the image in Azul, run:
-(cd ../azul && azul_docker_elasticsearch_version=7.17.15-6 azul_docker_registry=localhost:5000/ make test)
+# To test the image in Azul, you will need to temporarily modify Azul's
+# `environment.py` to set the appropriate `azul_docker_images` value using the
+# full image name noted above (starting with "localhost"). With this change in
+# place, run:
+_refresh
+make image_manifests.json
+azul_docker_registry="" make test
+# Finally, back in your working tree for this repository:
 make stop_registry
 ```

